massa-models/src/operation.rs

```diff
@@ -831,16 +788,11 @@ impl WrappedOperation {
             OperationType::Transaction { amount, .. } => *amount,
             OperationType::RollBuy { roll_count } => roll_price.saturating_mul_u64(*roll_count),
             OperationType::RollSell { .. } => Amount::zero(),
-            OperationType::ExecuteSC {
-                max_gas, gas_price, ..
-            } => gas_price.saturating_mul_u64(*max_gas),
-            OperationType::CallSC {
-                max_gas,
-                gas_price,
-                coins,
-                ..
-            } => gas_price
-                .saturating_mul_u64(*max_gas)
+            OperationType::ExecuteSC { max_gas, .. } => {
+                Amount::from_str(&max_gas.to_string()).unwrap_or(Amount::default())
+            }
+            OperationType::CallSC { max_gas, coins, .. } => Amount::from_str(&max_gas.to_string())
```

Comment:
here we should return just the `coins` property of CallSC

also, going through strings is inefficient